File: Year II/First Semester/MAP/Seminar/seminar1/seminar1/src/model/StackContainer.java
package model;

import java.util.ArrayList;

public class StackContainer implements Container {

    private ArrayList<Task> tasks;

    public StackContainer() {
        this.tasks = new ArrayList<>();
    }

    @Override
    public Task remove() {
        if(isEmpty())
            return null;
        else {
            return tasks.remove(size() - 1);
        }
    }

    @Override
    public void add(Task task) {
        tasks.add(task);
    }

    @Override
    public int size() {
        return tasks.size();
    }

    @Override
    public boolean isEmpty() {
        return tasks.isEmpty();
    }
}

File: Year II/First Semester/MAP/Laborator/Laborator9/ConsoleApp1/ConsoleApp1/Model/StackContainer.cs
namespace ConsoleApp1.Model;

public class StackContainer: AbstractTaskContainer
{
    public override Task Remove()
    {
        if(Tasks.Count == 0)
            throw new InvalidOperationException("Stack container is empty");
        
        Task removedTask = Tasks[Tasks.Count - 1];
        Tasks.RemoveAt(Tasks.Count - 1);
        
        return removedTask;
    }
}